valkey-k8s-operator

```diff
@@ -0,0 +1,267 @@
+      path: /opt/valkey/modules/libvalkey_bloom.so
+```
+
+Operator Behavior
+
+- Deploys Bitnami Valkey chart under the hood.
+- Supports 2 modes:
+  - Standalone: 1 primary, optional replicas.
+  - Sentinel: quorum-based HA with automatic failover.
+- Exposes Valkey connection info as a Kubernetes `Secret`.
+- Handles TLS setup via cert-manager or self-signed certs.
+- Optionally loads custom config via ConfigMap.
+- Optional metrics sidecar for Prometheus scraping.
+
+Configuration
+
+| Feature             | Config Options                                                                 |
+|---------------------|--------------------------------------------------------------------------------|
+| Deployment Mode     | `spec.sentinel.enabled: true/false`                                           |
+| TLS                 | `spec.tls.enabled: true` with cert-manager integration                         |
+| Persistence         | `spec.persistence.type: rdb/aof/both`, `storageClass`, `size`                 |
+| Modules             | `spec.modules`: list of `.so` modules to include                               |
+| Monitoring          | `spec.metrics.enabled: true`                                                   |
+| Custom Config       | `spec.config.configMapName`                                                   |
+| Pod Customization   | node selectors, tolerations, resource limits, securityContext, etc.           |
+
+Secret Output
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: valkey-test-binding
+type: Opaque
+stringData:
+  host: valkey-test.svc
+  port: "6379"
+  password: ********
+  sentinelHost: valkey-test.svc
+  sentinelPort: "26379"
+  tlsEnabled: "true"
+```
+
+Authentication and Authorization
+
+- No RBAC-specific changes are needed outside standard Kubernetes role management.
+- Valkey credentials are stored in Kubernetes Secrets.
+- TLS certs issued via cert-manager will follow existing RBAC rules.
+
+RDB
+
+- Persistence modes include:
+  - `rdb`: periodic snapshotting
+  - `aof`: append-only logging
+  - `both`: hybrid approach (default for production)
+
+Configuration (Runtime)
+
+- Users can override default Helm values using CR fields or a `valkey.conf` file via ConfigMap.
+- Sensitive data like passwords are not stored in the config map and are auto-generated and stored in Secrets.
+
+Keyspace notifications
+
+N/A – this Operator does not modify Valkey internals; only orchestrates deployment.
+
+Cluster mode
+
+- Sentinel mode uses quorum-based failover.
+- Standalone mode supports scaling replicas.
+
+Module API
+
+- Users can opt-in to modules like `Search`, `Bloom`, and `JSON` by including them in the container image and enabling via `valkey.conf`.
+
+Replication
+
+- Sentinel handles failover logic.
+- Operator ensures correct replica configurations based on `spec.replicas`.
+
+Dependencies
+
+- Bitnami Valkey Helm Chart
+- cert-manager (optional)
+- Prometheus Operator (optional)
+
+Benchmarking
```

Comment:
These parts of the template doesn't make sense for an operator. They are more intended for an RFC describing a new feature in Valkey itself, a feature that extends the module API for example. Let's ignore much of these sections of template and look at the README instead to see what's expected from an RFC.